Gems/Multiplayer/Code/Include/Multiplayer/Components/NetworkHierarchyRootComponent.h

```diff
@@ -57,6 +57,8 @@ namespace Multiplayer
         void BindNetworkHierarchyLeaveEventHandler(NetworkHierarchyLeaveEvent::Handler& handler) override;
         //! @}
 
+        const AZStd::vector<AZ::Entity*>& GetHierarchicalEntitiesRef() const;
```

Comment:
Are you looking to avoid copies here? And not use AZStd::vector<AZ::Entity*> GetHierarchicalEntities() const?
Be careful with keeping a reference as the list may change under you. The current size is under 16 entities or less, so this shouldn't be a concern.